clio-dataaccess-model/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/AutoElasticsearchIndex.scala

```diff
@@ -14,59 +15,106 @@ import scala.reflect.runtime.universe.Type
-      case tpe if tpe =:= typeOf[Option[Boolean]]   => booleanField _
-      case tpe if tpe =:= typeOf[Option[Int]]       => intField _
-      case tpe if tpe =:= typeOf[Option[Long]]      => longField _
-      case tpe if tpe =:= typeOf[Option[Float]]     => floatField _
-      case tpe if tpe =:= typeOf[Option[Double]]    => doubleField _
-      case tpe if tpe =:= typeOf[Option[String]]    => keywordField _
-      case tpe if tpe <:< typeOf[Option[EnumEntry]] => keywordField _
+    fieldType match {
+      case tpe if tpe =:= typeOf[Boolean]           => booleanField
+      case tpe if tpe =:= typeOf[Int]               => intField
+      case tpe if tpe =:= typeOf[Long]              => longField
+      case tpe if tpe =:= typeOf[Float]             => floatField
+      case tpe if tpe =:= typeOf[Double]            => doubleField
+      case tpe if tpe =:= typeOf[String]            => keywordField
+      case tpe if tpe =:= typeOf[Symbol]            => keywordField
+      case tpe if tpe <:< typeOf[EnumEntry]         => keywordField
+      case tpe if tpe =:= typeOf[OffsetDateTime]    => dateField
+      case tpe if tpe =:= typeOf[UpsertId]          => keywordField
+      case tpe if tpe =:= typeOf[Option[Boolean]]   => booleanField
+      case tpe if tpe =:= typeOf[Option[Int]]       => intField
+      case tpe if tpe =:= typeOf[Option[Long]]      => longField
+      case tpe if tpe =:= typeOf[Option[Float]]     => floatField
+      case tpe if tpe =:= typeOf[Option[Double]]    => doubleField
+      case tpe if tpe =:= typeOf[Option[String]]    => keywordField
+      case tpe if tpe =:= typeOf[Option[Symbol]]    => keywordField
+      case tpe if tpe <:< typeOf[Option[EnumEntry]] => keywordField
+      case tpe if tpe =:= typeOf[Option[UUID]]      => keywordField
+      case tpe if tpe =:= typeOf[Option[URI]]       => keywordField
       case tpe if tpe =:= typeOf[Option[OffsetDateTime]] =>
-        dateField _
+        dateField
       /*
        * Elasticsearch has no notion of an array field; instead, any
        * field can hold 1+ values of the same type.
        */
-      case tpe if tpe <:< typeOf[Option[Seq[String]]] => keywordField _
+      case tpe if tpe <:< typeOf[Option[Seq[String]]] => keywordField
+      case tpe if tpe <:< typeOf[Option[Seq[URI]]]    => keywordField
       case _ =>
-        throw new IllegalArgumentException(
-          s"No support for $fieldName: $fieldType"
-        )
+        (fieldName: String) =>
+          throw new IllegalArgumentException(
+            s"No support for $fieldName: $fieldType"
+          )
+    }
+  }
+
+  private def stringToDefinitionV2(
+    fieldType: Type
+  ): (String => FieldDefinition) = {
+    import scala.reflect.runtime.universe.typeOf
+
+    /*
+     * For Strings, create a top-level text field for searches,
+     * with a nested keyword field for sorting / aggregations.
+     */
+    val textFieldWithKeyword =
+      textField(_: String).fields(keywordField("exact"))
```

Comment:
This was mentioned as common practice in the Elasticsearch docs and seemed like a useful idea, but I can remove it if we don't want it.